src/test/java/com/metamx/common/StringUtilsTest.java

```diff
@@ -26,6 +26,34 @@
  */
 public class StringUtilsTest
 {
+  // copied from https://github.com/druid-io/druid/pull/2612
+  public final static String[] TEST_STRINGS = new String[]{
+      "peach", "péché", "pêche", "sin", "",
+      "☃", "C", "c", "Ç", "ç", "G", "g", "Ğ", "ğ", "I", "ı", "İ", "i",
+      "O", "o", "Ö", "ö", "S", "s", "Ş", "ş", "U", "u", "Ü", "ü", "ä",
+      "\uD841\uDF0E",
+      "\uD841\uDF31",
+      "\uD844\uDC5C",
+      "\uD84F\uDCB7",
+      "\uD860\uDEE2",
+      "\uD867\uDD98",
+      "\u006E\u0303",
+      "\u006E",
+      "\uFB00",
+      "\u0066\u0066",
+      "Å",
+      "\u00C5",
+      "\u212B"
+  };
+
+  @Test
+  public void binaryLengthAsUTF8Test() throws UnsupportedEncodingException
+  {
+    for (String string : TEST_STRINGS) {
+      Assert.assertEquals(StringUtils.toUtf8(string).length, StringUtils.binaryLengthAsUTF8(string));
+    }
+  }
+
```

Comment:
let's also add a tests where the length are not equal to clarify that this is the expected behavior